utils/slices.go

```diff
@@ -32,3 +32,13 @@ func Filter[T any](slice []T, f func(T) bool) []T {
 
 	return result
 }
+
+func IndexFunc[T comparable](slice []T, f func(T) bool) int {
```

Comment:
this function is going to be available in stdlib since go1.21, will remove when we upgrade go version